includes/class-wc-payment-gateway-wcpay.php

```diff
@@ -27,7 +27,7 @@ class WC_Payment_Gateway_WCPay extends WC_Payment_Gateway_CC {
 	 *
 	 * @type string
 	 */
-	const GATEWAY_ID = 'woocommerce_payments';
```

Comment:
Good call!

Even given this change, I am still wondering if repurposing `WC_Payment_Gateway_WCPay` to house the settings won't cause problems with plugins that build upon WooCommerce Payments.

I imagine a plugin that would like to e.g. check if the (previously only) WCPay gateway is registered will depend on the gateway's class name or ID.

Since you have used the ID `woocommerce_payments` on the new CC class I think the IDs will not be a problem. But if the gateway that has the class name `WC_Payment_Gateway_WCPay` does not provide some functionality it used to provide, this could cause problems.

I have no idea how many (if any) such plugins exist. If this turns out to be an issue though, settings management could be extracted to a new class, and `WC_Payment_Gateway_WCPay` could proxy calls to the new CC class, with each method marked as deprecated.

Reply:
Another way to fix this issue in the third party plugin is to update their `extends WC_Payment_Gateway_WCPay` to `extends CC_Payment_Gateway`. Then the existing credit card based functionality would be the same.

Reply:
Right! I'm just worried about what will happen in the time span between updating WCPay and updating the third-party plugin.

Reply:
Thanks for bringing it up!